s3-native-client/src/test/java/com/amazonaws/s3/S3NativeClientTest.java

```diff
@@ -160,6 +168,115 @@ public void onException(final CrtRuntimeException e) {
         }
     }
 
+    private class CancelTestData<T>
+    {
+        public int ExpectedPartCount;
+        public int PartCount;
+        public CompletableFuture<T> ResultFuture;
+    }
+
+    @Test
+    public void testGetObjectCancel() {
+        Assume.assumeTrue(System.getProperty("NETWORK_TESTS_DISABLED") == null);
+
+        try (final EventLoopGroup elGroup = new EventLoopGroup(9);
+                final HostResolver resolver = new HostResolver(elGroup, 128);
+                final ClientBootstrap clientBootstrap = new ClientBootstrap(elGroup, resolver);
+                final CredentialsProvider provider = getTestCredentialsProvider()) {
+
+            Exception exceptionResult = null;
+
+            final S3NativeClient nativeClient = new S3NativeClient(REGION, clientBootstrap, provider, 64_000_000l,
+                    100.);
+
+            final CancelTestData<GetObjectOutput> testData = new CancelTestData<GetObjectOutput>();
+            testData.ExpectedPartCount = 1;
+
+            try {
+                testData.ResultFuture =
+                         nativeClient.getObject(GetObjectRequest.builder().bucket(BUCKET).key(GET_OBJECT_KEY).build(),
+                                                new ResponseDataConsumer<GetObjectOutput>() {
+
+                            @Override
+                            public void onResponse(GetObjectOutput response) {
+                                assertNotNull(response);
+                            }
+
+                            @Override
+                            public void onResponseData(ByteBuffer bodyBytesIn) {
+                                ++testData.PartCount;
+
+                                if (testData.PartCount == testData.ExpectedPartCount) {
```

Comment:
Is there any value in also cancelling earlier or is this sufficient coverage?

Reply:
We have a lot of cancel tests at the C layer, so I'm leaning toward this being sufficient.

Reply:
Added a couple of additional tests.